### DISCUSSION TITLE
Zigbee2mqtt, Dongle Sonoff Zigbee 3.0 e Sensori Apertura finestre Xiaomi/Aqara (gen1) che perdono connessione

### DISCUSSION THREAD

**Discussion by lcorsini** [indomus/forum#2593] [ZigBee, Z-Wave, Thread, Matter]
Buongiorno, 
Sto migrando, usando zigbee2mqtt i miei dispositivi, da un coordinator Raspbee (versione 1) verso la dongle di Sonoff (ZBDongle-E) e sono incappato in questo problema, apparentemente bizzarro, ho sparsi per casa i sensori in oggetto, il pairing è andato tutto bene, tuttavia facendo una mappa dei dispositivi dalla ui di z2m mi sono accorto che pian piano, perdono connessione, restano apparentemente online, ma non aggiornano piu il loro stato.
con il precedente coordinator non succedeva... a qualcuno è capitata questa situazione?

**Comment by snakuzzo** [comment]
Tutti i sensori a batteria vanno in deep sleep....è scontato che perdano la connettività.
Si riattivano solo al cambio stato. Quella mappa lascia il tempo che trova, così come lascia il tempo che trova quella che utilizzavi prima.
Quello che ti viene mostrato su quelle schifezze di mappe è basato sulla logica del software che gira sotto...e le logiche potrebbero essere diverse.
Ribadisco...lascia perdere le mappe...Il problema ponitelo solo  se vedi che i device non funzionano.

**Reply by lcorsini** [reply]
me lo sto ponendo proprio perchè non funzionano :) altrimenti non sarei venuto qua.
il fatto che vadano in deep sleep mi sta bene, ma, come ho scritto sopra, non vengono piu trovati da z2m e di conseguenza non si registra piu il cambio di stato, che resta fermo all'ultima volta in cui erano attivi
la mappa farà anche schifo, ma quando li resetto e si riaccoppiano la mappa li mostra correttamente connessi e ricevo gli eventi e gli attributi del dispositivo.

I device funzionano, sia perchè con la raspbee hanno funzionato per anni, sia perchè effettivamente appena connessi mandano regolarmente i cambi di stato (aperto/chiuso)

**Reply by snakuzzo** [reply]
Allora controlla i log di zigbee2mqtt e controlla la cosa del broker mqtt. Questo perché, se ci dobbiamo basare solo sul "non funzionano", ci vorrebbe la sfera di cristallo.
Se non trovi nulla, prova a verificare anche tra le issue sul progetto zigbee2mqtt.
Se proprio non ne esci, prova a riconfigurare la rete zigbee, magari verifica bene pure il canale scelto

**Reply by lcorsini** [reply]
i log non dicono nulla, semplicemente il dispositivo smette di mandare aggiornamenti.
l'unica issue relativa a questi dispositivi parla di perdita del segnale con certi repeater (https://www.zigbee2mqtt.io/devices/MCCGQ01LM.html#xiaomi-mccgq01lm), ma sulla mia rete al momento non ce ne sono, c'è solo il coordinator.
(e nella precedente tra l'altro c'era un repeater incompatibile ma funzionavano tutti)
non ho trovato specifiche incompatibilità con l'antenna in oggetto ma a questo punto qualche dubbio mi viene.
ho già cambiato canale, cambiato tutte le batterie e riconfigurato varie volte i dispositivi

**Reply by snakuzzo** [reply]
Credo che quasi nessuno utilizzi più quel coordinator. Già anni fa era poco usata, immagina ora.
Più di consigliarti l'utilizzo di un coordinator più recente (vedi ZBDongle di ITEAD) non saprei cosa dirti.

**Reply by lcorsini** [reply]
eh peccato che sia proprio la ZBDongle di ITEAD è quella con cui ho problemi con questi device (eppure nel post originale l'ho scritto)
No seriamente la raspbee è spettacolare per quanto abbia funzionato bene (ripeto tutti i device che ho in casa funzionano perfettamente da anni con quella)
Sto migrando alla ZBDongle solo per una questione di reperibilità (la vers. 1 non è piu in produzione), costi (costa cmq di più) e compatibilità (la raspbee è limitata essendo connessa via gpio) ne ho comprate 2 per avere un backup, ma se con i dispositivi xiaomi non funzionano la cosa potrebbe costarmi parecchio e a sto punto compro 2 conbee

**Reply by snakuzzo** [reply]
Ops...avevo rimosso che stavi migrando a z2m con la ZBDongle-E.
Ricorda che, proprio per questo adapter, per zigbee2mqtt il supporto è ancora "sperimentale" (l'abbiamo scritto). Quindi potrebbe dipendere anche da questo.

**Reply by lcorsini** [reply]
può darsi, ho aperto una issue a z2m per vedere se lato loro ci sono verifiche che posso fare
ho notato anche che non sono all'ultimissima versione del firmware della dongle (la v1.0.1) che sonoff indica come stabile, ma alla precedente (build 297) magari provo anche ad aggiornarla nel frattempo

**Reply by lcorsini** [reply]
dopo una giornata di prove, è chiaro che la dongle e questi sensori non vanno d'accordo.
Nel frattempo ho trovato una soluzione, aggiungendo dei dispositivi router (delle prese smart tuya e dei relè sonoff ZBMINI che tanto devono far parte della rete quando avrò tempo di installarli a tutti i pulsanti delle luci) e facendo fare il pairing su questi ultimi e non sul coordinator la connessione sembra rimanere attiva.
domani farò un test con una prolunga usb piu lunga per capire se è un problema di interferenze o range della dongle giusto per togliermi il dubbio, ma a questo punto penso che forzerò la connessione sui router, quando nella rete ce ne saranno una decina non dovrei avere piu problemi

**Reply by KC-inDomus** [reply]
se ti può consolare, io ne ho uno (di sensore xiaomi porte/finestre) e lo uso proprio con z2m e zbdongle-e. si disconnette anche lui, ma funziona perfettamente lo stesso.

**Reply by lcorsini** [reply]
invece il mio dopo un'ora (probabilmente il tempo di andare in deep sleep) smette completamente di mandare eventi al coordinator, lo perde proprio (ho notato che quando è disconnesso la lucina blu lampeggia quando apro/chiudo il sensore)
facendo il pairing con un router invece funziona, tra l'altro ho visto oggi che rimuovendo uno dei 2 router che ci sono (al momento la mia rete zigbee è in cantiere dato che devo montare i sonoff nelle scatole, e ne avevo connesso uno volante a cui ho accoppiato uno dei sensori xiaomi) il sensore xiaomi si è riagganciato senza problemi all'altro router presente nella rete...
devo ancora fare la prova con la prolunga usb per vedere se è fosse un problema di interferenze, ma a sto punto se il pairing con i router funziona me li tengo così

**Reply by developerGM** [reply]
> dopo una giornata di prove, è chiaro che la dongle e questi sensori non vanno d'accordo. Nel frattempo ho trovato una soluzione, aggiungendo dei dispositivi router (delle prese smart tuya e dei relè sonoff ZBMINI che tanto devono far parte della rete quando avrò tempo di installarli a tutti i pulsanti delle luci) e facendo fare il pairing su questi ultimi e non sul coordinator la connessione sembra rimanere attiva. domani farò un test con una prolunga usb piu lunga per capire se è un problema di interferenze o range della dongle giusto per togliermi il dubbio, ma a questo punto penso che forzerò la connessione sui router, quando nella rete ce ne saranno una decina non dovrei avere piu problemi

ciao, sto avendo gli stessi problemi. Dopo 1 giorno mi ritrovo dei sensori alimentati a batteria che si disconnettono. Durante il pairing si collegano alle prese (router) della biticino ma vedo se sono poco stabili. I dispositivi che invece si collegano alle prese smart invece rimangono connessi. Come hai fatto a forzare il pairing su determinate prese? Perchè anche mettendomi vicino, non le prende da solo.

**Comment by lcorsini** [comment]
Allora, come ultimo test ho messo la dongle fuori dall'armadietto con una prolunga più lunga.
ed ho riprovato a fare il pairing direttamente con il coordinator.
dopo 30 minuti ho già perso il device allo stesso modo, tra l'altro ho notato che col coordinator non vengono riportati i valori di voltaggio.
quindi è evidente che questi dispositivi non vanno molto daccordo con la ZBDongle-E probabilmente perchè, come riporta anche il sito di zigbee2mqtt non implementano correttamente tutto lo standard zigbee.
la cosa strana è il fatto che sul sito riportano possibili problemi con dispositivi router, mentre nel mio caso avviene il contrario :D

**Comment by faustozago** [comment]
A me a volte succede la stessa cosa ( sensori che dalla mappa sono proprio sconnessi, con LQi=0 o addirittura senza linea che li collega) , ma nella realta'  funzionano normalmente .
Io ho sonoff dongle-P

**Reply by lcorsini** [reply]
la tua è differente e credo maggiormente supportata, cmq nel mio caso, quello visivo non sarebbe un problema, il fatto è che proprio si isolano dopo circa un'ora e non si riconnettono piu, accoppiandoli con dei router (in questo caso 2 prese smart tuya e dei sonoff zbmini2) invece restano collegati aggiornando correttamente il loro stato(e appaiono anche collegati nella mappa)

**Reply by lcorsini** [reply]
di base penso che andrò poi a sostituire questi con qualcosa di piu recente e conforme al protocollo zigbee, leggevo che i primi sensori aqara/xiaomi non sono tanto conformi
dovrei provare a rimettere in piedi il loro gateway per vedere se ci sono aggiornamenti

**Reply by diegoflyer** [reply]
@lcorsini ciao! ho un problema simile ma non riesco a capire come fare il pairing su un router al posto del coordinatore, dove trovo le indicazioni?

**Reply by KC-inDomus** [reply]
ti basta selezionare in alto nel menu a tendina del pairing quale elemento utilizzare

**Comment by giannicarabelli** [comment]
provato a cambiare frequenza? 
la 2.4Ghz è piena... magari nel precedente setup eri su altro canale.

**Reply by lcorsini** [reply]
si si, cambiare canale è la prima cosa che ho fatto (anche perchè tra wifi e il bridge hue di traffico ce n'è tanto) ma non cambia nulla, e sono gli unici device a batteria che hanno questo problema altri sensori quali temperatura tuya e aqara e pulsanti aqara non hanno nessun tipo di problema

**Comment by IvanDeMarco** [comment]
@lcorsini Ciao, io ho il tuo stesso problema, hai risolto?

Grazie!

**Reply by lcorsini** [reply]
parzialmente, come dicevo sopra, se faccio il pairing direttamente con il coordinator hanno il problema, se invece faccio il pairing (dall'interfaccia web di z2m) con un router presente nella mia rete (nel mio caso un paio di smart plug tuya e dei relè sonoff zbmini2) con loro la connessione viene mantenuta attiva e aggiornano correttamente il loro stato

**Reply by snakuzzo** [reply]
Questo è molto strano. Hai provato ad aprire una issue sul progetto di zigbee2mqtt ?

**Reply by lcorsini** [reply]
si, [aperta](https://github.com/Koenkk/zigbee2mqtt/issues/18698), ignorata e chiusa :D

**Reply by snakuzzo** [reply]
ottimo :D

**Reply by lcorsini** [reply]
purtroppo non ci sono molte info anche mettendo i log verbose, il device dopo 1 ora va in deep sleep e con la dongle perde la connessione e non aggiorna piu lo stato, mentre i router (sia tuya che sonoff) mantengono il device connesso che quindi al risveglio aggiorna lo stato

**Reply by trebestie** [reply]
Ho avuto lo stesso problema e  ho risolto tornando al dongle precedente. Con quello non c'era verso.

**Reply by lcorsini** [reply]
non potendo tornare al raspbee non avevo molte alternative, fortunatamente funzionano con i sonoff
poi siccome sono gli unici device che danno problemi pian piano li dismetterò

**Comment by KC-inDomus** [comment]
per inciso, io ho un Xiaomi MCCGQ01LM visivamente "disconnesso" dalla rete zigbee, ma che funziona comunque perfettamente.

**Reply by lcorsini** [reply]
il disconnesso non sarebbe un problema, ho anche io altri dispositivi EndDevice che sulla mappa spesso risultano disconnessi ma in realtà aggiornano il loro stato

questi invece, come scrivevo sopra, dopo un ora smettono di aggiornarsi se faccio il pairing direttamente con il coordinator, mentre forzando il pairing con  un qualsiasi router della rete zigbee restano connessi e aggiornati

**Comment by trebestie** [comment]
A gennaio ho comprato il dongle-e in sostituzione del vecchio cc2531.
A fronte di un significativo incremento della portata avevo disconnessioni
random di molti device, soprattutto vecchi Xiaomi/Aqara.
Ho fatto il reso e comprato il dongle-p. Ho dovuto mantenere gran parte dei
router che speravo di eliminare, ma adesso funziona tutto discretamente
bene.
Vedo che purtroppo il problema non è ancora stato risolto.


Il giorno mer 18 ott 2023 alle ore 10:26 Luca Corsini <
***@***.***> ha scritto:

**Reply by lcorsini** [reply]
eh il problema è che ormai, oltre ad averne comprati 2 di dongle-e, dovrei rifare tutta la rete, ho iniziato a montare gli zbmini e quelli vanno nelle scatole dietro ai pulsanti, sarebbe una bella rottura smontare tutto, e solo per quella manciata di sensori xiaomi forse non ne vale la penaè anche vero che scavando un pò nella documentazione le dongle plus sembrano molto meglio supportate, con la possibilità di fare i backup

ecco adesso finisce che ne compro un paio di dongle-p  pensavo le -e fossero meglio ma a sto punta mi sa che ho fatto una minchiata a prenderle

**Reply by snakuzzo** [reply]
Non è una questione di meglio o peggio.
Sono due device con due architetture diverse. La E ti permette di gestire sia i device zigbee che thread.
Poi ci può stare che ci siano problemi. Vero è che non tutti hanno i problemi di cui parlate voi

**Reply by lcorsini** [reply]
Qua ci sono 2 questioni.
le ezsp sono ancora dichiarate experimental da z2m
mentre quelle basate su CC2XXX sono pienamente supportate, tra l'altro, è supportato anche il backup e sostituzione della dongle.
L'altra questione è che i dispositivi di cui stiamo parlando sono roba o davvero vecchia, o cmq non totalmente rispettosa dello standard zigbee.

personalmente ho preferito il maggior supporto (thread siamo ancora veramente agli inizi) e ho rimpiazzato anzi il coordinator

**Reply by snakuzzo** [reply]
Scelta, nel tuo caso, pienamente condivisibile

**Reply by KC-inDomus** [reply]
> le ezsp sono ancora dichiarate experimental da z2m
mentre quelle basate su CC2XXX sono pienamente supportate, tra l'altro, è supportato anche il backup e sostituzione della dongle

sperimentali vuol dire poco: per lo più che lo sviluppo di zigbee2mqtt non eroga specificamente firmware per quelle componenti _e quindi_ sono considerate sperimentali. non per questo non sono perfettamente funzionanti

**Reply by lcorsini** [reply]
sperimentali per le faq di z2m significa:
The adapters below are experimental, don't use these if you want a stable setup.

e non sono perfettamente funzionanti per i problemi che stiamo avendo in questo thread, tant'è che adesso sembra ci sia indicata una soluzione al problema di perdita di connessione

It's recommended to flash a different firmware using this [web flasher](https://dialedin.com.au/blog/sonoff-zbdongle-e-rcp-firmware). The pre-flashed firmware ncp-uart-sw_EZNet6.10.3_V1.0.1 has issues with loosing connection to various devices (typically Xiaomi/Aqara)

**Reply by KC-inDomus** [reply]
no guarda, centinaia di persone usano conbee, oppure zbdongle-e con firmware ncp senza mezzo problema, eppure sono "sperimentali". il fatto che loro lo definiscano sperimentale è per i motivi che ti ho scritto sopra 

p.s. *"che stai avendo in questo thread"

**Reply by snakuzzo** [reply]
E comunque, recuperate il firmware da [questo repository](https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/zbdonglee).
Trovate:
- NCP (per la parte solo zigbee)
- OT (Open Thread per la parte SOLO Thread)
- RCP (per utilizzare sia Zigbee che Thread con specifico addon/software multiprotocol)

**Reply by lcorsini** [reply]
beh se guardi quella pagina raspbee/conbee etc non stanno piu sotto experimental ma genericamente sotto other  mentre la dongle-e invece è ancora li

poi che funzioni al 99% bene è un altro discorso, così come che il massimo delle feature ci siano implementate solo con zstack

pure io usavo raspbee prima senza nessun problema con i sensori aqara

**Reply by lcorsini** [reply]
le dongle escono con l'ultimo NCP che ha ancora questo problema, probabilmente in quello RCP hanno fatto qualche fix a giudicare da quanto scrivono in documentazione (il link che ho messo prima del web flasher)

**Reply by KC-inDomus** [reply]
conbee finalmente l'hanno tolta perché era davvero una barzelletta. zbdongle-e, ripeto, funziona benissimo con ncp. poi che loro vogliano battezzarla "experimental" per i loro motivi, liberi. ma la verità empirica ad oggi è un'altra.

**Reply by lcorsini** [reply]
beh funziona benissimo tranne che con certi device, e non lo sto dicendo solo io in questo thread, ribadisco hanno riconosciuto che quelle dongle hanno problemi con alcuni device e consigliano di passare al firmware RCP

It's recommended to flash a different firmware using this [web flasher](https://dialedin.com.au/blog/sonoff-zbdongle-e-rcp-firmware). The pre-flashed firmware ncp-uart-sw_EZNet6.10.3_V1.0.1 has issues with loosing connection to various devices (typically Xiaomi/Aqara)

**Reply by snakuzzo** [reply]
@lcorsini per favore, leggi il mio commento con il link al repository...prova quel firmware

**Reply by lcorsini** [reply]
@snakuzzo ho visto il link, è lo stesso repo che linkano nella guida di z2m solo che la applicano quello RCP
(che a leggere, oltre ad implementare il multiprocotol, sposta parte dello stack sulla macchina, lasciando alla dongle solo la gestione radio)
Lo proverò per vedere se effettivamente risolve anche mantenendolo NCP :)

**Reply by snakuzzo** [reply]
Sì, quello intendevo...provare il firmware NCP

**Comment by diegoflyer** [comment]
@lcorsini  Ciao! come hai fatto a forzare la connessione sui router? Io riesco a fare il pairing normalmente ma poi si perde qualche segnale che dovrebbe arrivare almeno ogni ora.
Hai usato la funzione bind?

**Comment by lcorsini** [comment]
Ho abilitato il pairing solo su uno dei router dalla web ui
Dalla mappa ti risulteranno cmq disconnessi dopo un po' perché sono end
device, ma vedrai che anche dopo un ora se apri la porta segnaleranno il
loro stato. PS anche con i router dicono che non siano perfettamente
compatibili
Io ho testato con delle smart plug tuya e osram, oltre che degli zbmini2
